orchagent/neighorch.cpp

```diff
@@ -1420,6 +1420,17 @@ void NeighOrch::doVoqSystemNeighTask(Consumer &consumer)
                     {
                         SWSS_LOG_NOTICE("VOQ encap index updated for neighbor %s", kfvKey(t).c_str());
                         it = consumer.m_toSync.erase(it);
+
+                       /* Remove remaining DEL operation in m_toSync for the same neighbor.
+                        * Since DEL operation is supposed to be executed before SET for the same neighbor
+                        * A remaining DEL after the SET operation means the DEL operation failed previously and should not be executed anymore
+                        */
+                       auto rit = make_reverse_iterator(it);
+                       while (rit != consumer.m_toSync.rend() && rit->first == key && kfvOp(rit->second) == DEL_COMMAND)
+                       {
```

Comment:
i thought m_toSync will only have one entry for each neighbor, i didn't know if we have multiple entry for the same neighbor, just curious.

Reply:
why this problem is voq neighbor specificially? for normal neighbor, why we do not have such problem?

Reply:
> i thought m_toSync will only have one entry for each neighbor, i didn't know if we have multiple entry for the same neighbor, just curious.

m_toSync is type of std::multimap, so it allows multiple actions (like DEL and SET) for the same neighbor.

Reply:
> why this problem is voq neighbor specificially? for normal neighbor, why we do not have such problem?

This problem is not voq specific. The same change/fix exists for local neighbor too. The PR is to apply the same fix for voq.